src/openassetio-python/tests/bridge/python/test_converter.cpp

```diff
@@ -124,26 +128,28 @@ SCENARIO("Casting to a C++ object binds object lifetime") {
     PyObject* pyTraitsData = pyInstance.release().ptr();
     CHECK(Py_REFCNT(pyTraitsData) == 1);
 
-    WHEN("The python object is converted to a C++ object") {
+    WHEN("Python object is converted to a C++ object") {
       TraitsDataPtr traitsData =
           openassetio::python::converter::castFromPyObject<TraitsData>(pyTraitsData);
 
-      THEN("The python reference count has been increased") {
-        CHECK(Py_REFCNT(pyTraitsData) == 2);
+      THEN("Python reference is obtained") { CHECK(Py_REFCNT(pyTraitsData) == 2); }
+
+      AND_WHEN("C++ reference is destroyed") {
+        traitsData.reset();
+
+        THEN("Python reference is released") { CHECK(Py_REFCNT(pyTraitsData) == 1); }
       }
     }
-    AND_WHEN("The c++ object falls out of scope") {
-      THEN("The python reference count is reduced") { CHECK(Py_REFCNT(pyTraitsData) == 1); }
-    }
```

Comment:
[Question] Ah was this the scope error? Does `AND_X` not cause the sections to be run sequentially? I guess not. Does that mean `AND` is just flavour, with no mechanical effect?

Reply:
Yeah all those `THEN`, `AND_GIVEN` etc etc are just synonyms for `SECTION`, only difference being how they render when there's an error.